compiler/rustc_session/src/options.rs

```diff
@@ -59,18 +60,187 @@ macro_rules! hash_substruct {
     };
 }
 
+/// Extended target modifier info.
+/// For example, when external target modifier is '-Zregparm=2':
+/// Target modifier enum value + user value ('2') from external crate
+/// is converted into description: prefix ('Z'), name ('regparm'), tech value ('Some(2)').
+pub struct ExtendedTargetModifierInfo {
+    /// Flag prefix (usually, 'C' for codegen flags or 'Z' for unstable flags)
+    pub prefix: String,
+    /// Flag name
+    pub name: String,
+    /// Flag parsed technical value
+    pub tech_value: String,
+}
+
+/// A recorded -Zopt_name=opt_value (or -Copt_name=opt_value)
+/// for ABI-changing or vulnerability-affecting options.
```

Comment:
```suggestion
/// which alter the ABI or effectiveness of exploit mitigations.
```
I don't think "vulnerability-affecting" is a good term for this. The usual term is "exploit mitigation".

Reply:
Fixed.